be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -343,6 +343,34 @@ Status SegmentIterator::_init_impl(const StorageReadOptions& opts) {
             _storage_name_and_type[i] = std::make_pair(col->name(), storage_type);
         }
     }
+
+    // find columns that definitely require reading data, such as functions that are not pushed down.
+    {
+        std::set<std::string> push_down_preds;
+        for (auto* pred : _col_predicates) {
+            if (!_check_apply_by_inverted_index(pred)) {
+                continue;
+            }
+            push_down_preds.insert(_gen_predicate_result_sign(pred));
```

Comment:
Pred may not be evaluated by index even if _check_apply_by_inverted_index(pred) return true due to skip index based on hit ratio. So push_down_preds.contains(column_sign) can not guarantee that the pred for column_sing can be skipped.

Reply:
This section only calculates which conditions definitely require data reading, the conditions that do not require data reading are computed in other logical calculations.